core/unit_test/TestTeamBasic.hpp

```diff
@@ -114,6 +114,75 @@ TEST(TEST_CATEGORY, team_reduce_large) {
   }
 }
 
+/*! \brief Test passing an aggregate to Kokkos::single in a parallel_for with
+           team policy
+*/
+template <typename ExecutionSpace>
+struct TestTeamForAggregate {
+  using range_policy_t = Kokkos::RangePolicy<ExecutionSpace>;
+  using team_policy_t  = Kokkos::TeamPolicy<ExecutionSpace>;
+  using member_t       = typename team_policy_t::member_type;
+  using memory_space   = typename ExecutionSpace::memory_space;
+  using results_type   = Kokkos::View<double*, memory_space>;
+
+  static constexpr double INIT_VALUE   = -1.0;
+  static constexpr double EXPECT_VALUE = 1.0;
+
+  struct Agg {
+    double d;
+  };
+  results_type results_;
+
+  TestTeamForAggregate() = default;
+  TestTeamForAggregate(const results_type& results) : results_(results) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const member_t& t) const {
+    Agg lagg;
+    lagg.d = INIT_VALUE;
+    Kokkos::single(
+        Kokkos::PerTeam(t), [&](Agg& myAgg) { myAgg.d = EXPECT_VALUE; }, lagg);
+    size_t i = t.league_rank() * t.team_size() + t.team_rank();
+    if (i < results_.size()) {
+      results_(i) = lagg.d;
+    }
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i, int& lNumErrs) const {
+    if (EXPECT_VALUE != results_(i)) {
+      ++lNumErrs;
+    }
+  }
+
+  static void run() {
+    // dummy objects to determine max team size
+    int maxTeamSize;
+    {
+      TestTeamForAggregate test;
+      maxTeamSize =
+          team_policy_t(1, 1).team_size_max(test, Kokkos::ParallelForTag());
+    }
+
+    for (int teamSize = 1; teamSize <= maxTeamSize; teamSize *= 2) {
```

Comment:
OpenMPTarget has
```
 RUN      ] openmptarget.team_parallel_single
4: OpenMPTarget backend requires a minimum of 32 threads per team.
4: .
```
so you would need to start this loop with 32 for that backend.

Reply:
https://github.com/kokkos/kokkos/issues/5458